README.md

```diff
@@ -762,6 +762,119 @@ val c: Int < Any =
+| Concatenation              | `concat`                                             | O(n)          | O(n)          |
+| Effectful map and filter   | `map`, `filter`, `collect`, `takeWhile`, `dropWhile` | O(n)          | O(n)          |
+| Effectful side effects     | `foreach`, `collectUnit`                             | O(n)          | O(n)          |
+| Effectful fold             | `foldLeft`                                           | O(n)          | O(n)          |
+| Copying to arrays          | `toArray`, `copyTo`                                  | O(n)          | O(n)          |
+| Other operations           | `flatten`, `changes`, `toSeq`, `toIndexed`           | O(n)          | O(n)          |
+
+When deciding between `Chunk` and `Chunks.Indexed`, consider the primary operations you'll be performing on the data. If you mainly need to `append` elements, `take` slices, or `drop` elements from the beginning or end of the sequence, `Chunk` is a good choice. Its `O(1)` complexity for these operations makes it efficient for such tasks.
+
+```scala
+import kyo._
+
+val a: Chunk[Int] = Chunks.init(1, 2, 3, 4, 5)
+
+// Efficient O(1) operations with Chunk
+val b: Chunk[Int] = a.append(6)
+val c: Chunk[Int] = a.take(3)
+val d: Chunk[Int] = a.dropLeft(2)
+```
+
+On the other hand, if you frequently need to access elements by index, `Chunks.Indexed` is the better option. It provides `O(1)` element access and supports `head` and `tail` operations, which are not available in `Chunk`.
+
+```scala
+import kyo._
+
+val a: Chunks.Indexed[Int] = 
+  Chunks.init(1, 2, 3, 4, 5).toIndexed
+
+// Efficient O(1) operations with Chunks.Indexed
+val b: Int = a(2)
+val c: Int = a.head
+val d: Chunks.Indexed[Int] = a.tail
+```
+
+Keep in mind that converting between `Chunk` and `Chunks.Indexed` is an `O(n)` operation, so it's best to choose the appropriate type upfront based on your usage patterns. However, calling `toIndexed` on a chunk that is already internally indexed is a no-op and does not incur any additional overhead.
+
+Here's an overview of the main APIs available in Chunk:
+
+```scala
+import kyo._
+
+// Creation
+val a: Chunk[Int] = Chunks.init(1, 2, 3)
+val b: Chunk[Int] = Chunks.initSeq(Seq(4, 5, 6))
+
+// Size and emptiness
+val c: Int = a.size
+val d: Boolean = a.isEmpty
+
+// Take and drop
+val e: Chunk[Int] = a.take(2)
+val f: Chunk[Int] = a.dropLeft(1)
+
+// Append and last
+val g: Chunk[Int] = a.append(4)
+val h: Int = a.last
+
+// Concatenation
+val i: Chunk[Int] = a.concat(b)
+
+// Effectful map and filter
+val j: Chunk[String] < Any = a.map(_.toString)
+val k: Chunk[Int] < Any = a.filter(_ % 2 == 0)
+
+// Effectful side effects
+val l: Unit < IOs = a.foreach(v => IOs(println(v)))
+
+// Effectful fold
+val m: Int < Any = a.foldLeft(0)(_ + _)
+
+// Copying to arrays
+val n: Array[Int] = a.toArray
+
+// Other operations
+val o: Chunk[Int] = Chunks.init(a, b)
+  .flatten
+val p: Chunk[Int] = Chunks.init(1, 1, 2, 3, 3)
+  .changes
```

Comment:
I think it would be helpful to comment the resulting value here (or any other non-trivial operator). That or we should find a mechanism to have runnable docs...